tests/Serializer/Type/ParserTest.php

```diff
@@ -79,6 +79,10 @@ public function validTypesProvider(): iterable
             'array<Foo\Bar, Baz\Boo>',
             $type('array', [['name' => 'Foo\Bar', 'params' => []], ['name' => 'Baz\Boo', 'params' => []]]),
         ];
+        yield [
+            'DateTime<null, null, array<\'Y-m-d\TH:i:s\', \'Y-m-d\TH:i:sP\'>>',
+            $type('DateTime', [null, null, ['name' => 'array', 'params' => ['Y-m-d\TH:i:s', 'Y-m-d\TH:i:sP']]]),
```

Comment:
They should be tested together to make sure it works as expected. 

Your current branch is not compatible with  https://github.com/schmittjoh/serializer/pull/1125. 
If you checkout   https://github.com/schmittjoh/serializer/pull/1125 and and test it with your code, it will fail.

`DateTime<null, null, ['Y-m-d\TH:i:s', 'Y-m-d\TH:i:sP'],`  will produce an array format that your `getDeserializationFormatsAndTimeZones`  will not be able to understand.

Reply:
Wouldn't be easier if the #1125 branch was merge into master ?